setup.py

```diff
@@ -40,6 +40,7 @@
     "dill>=0.3",
     "fastjsonschema>=2.10",
     "python-constraint>=1.4",
+    "python-dateutil>=2.8.0",
```

Comment:
I'm fine with adding dateutil to do the conversion, but we will need to add a release note about the new requirement if we're going to add it. That being said what was the format of the string we were passing to marshmallow before? If we're expecting a fixed form of input format, it might be covered by datetime natively, for example: 
https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp
https://docs.python.org/3/library/datetime.html#datetime.datetime.fromordinal
https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat
etc

Reply:
The old marshmallow code allowed any ISO format timestamp, and `fromisoformat()` is only available in 3.7+. Alternatively we can do `strptime()` with the format `qiskit-ibmq-provider` uses, since that's likely the only provider that returns `BackendProperties`. But I don't really like having code specifically for just a provider.